refinery/tool_manager/utils.py

```diff
@@ -507,3 +508,60 @@ def validate_tool_launch_configuration(tool_launch_config):
                 e
             )
         )
+
+
+def create_expanded_workflow_graph(galaxy_workflow_dict):
+    graph = nx.MultiDiGraph()
+    steps = galaxy_workflow_dict["steps"]
+
+    # iterate over steps to create nodes
+    for current_node_id, step in steps.iteritems():
+        # ensure node id is an integer
+        current_node_id = int(current_node_id)
+        # create node
+        graph.add_node(current_node_id)
+        # add node attributes
+        graph.node[current_node_id]['name'] = "{}:{}".format(
+            current_node_id,
+            step['name']
+        )
+        graph.node[current_node_id]['tool_id'] = step['tool_id']
+        graph.node[current_node_id]['type'] = step['type']
+        graph.node[current_node_id]['position'] = (
+            int(step['position']['left']), -int(step['position']['top'])
+        )
+        graph.node[current_node_id]['node'] = None
+    # iterate over steps to create edges (this is done by looking at
+    # input_connections, i.e. only by looking at tool nodes)
+    for current_node_id, step in steps.iteritems():
+        # ensure node id is an integer
+        current_node_id = int(current_node_id)
+        input_connections = step['input_connections'].iteritems()
+        for current_node_input_name, input_connection in input_connections:
+            parent_node_id = input_connection["id"]
+            # test if parent node is a tool node or an input node to pick the
+            # right name for the outgoing edge
+            if parent_node_id == 0:
```

Comment:
No more checking for `GALAXY_INPUT_TYPES`?

Reply:
No, that ended up being overkill